docs/reference/modules/node.asciidoc

```diff
@@ -1,32 +1,223 @@
+search and indexing traffic, as well as cluster related traffic
+(e.g., requesting the current cluster settings). Data nodes are the
+nodes that literally contain the data, so they are meant to handle
+search and indexing traffic, but cluster related traffic should be
+directed to master nodes whenever possible.
+
+To create a standalone data node:
+
+```yaml
+node.master: false
+node.data: true
+```
+
+NOTE: It is not enough to just set `node.data: true`. You must
+disable the master aspect, which is enabled by default.
+
+[float]
+[[client]]
+===== Client Node
+
+If you take away the ability to be able to handle master
+duties and take away the ability to hold data, then you are left
+with a node that can only route requests.
+
+Standalone client nodes provide large scale clusters a few
+benefits, by offloading some of the overhead associated with
+federated search and even bulk indexing. Client nodes join the
+cluster and receive the full cluster state, like every other
+node, and they use the cluster state to route requests to the
+appropriate place(s). In terms of searching and indexing, this
+is known as federated search and federated indexing.
+
+Unlike master nodes and data nodes, the entire purpose of
+client nodes is to handle and route requests. This means that it
+is good practice to send them any type of request, although they
+are generally expected to be used for data-related requests.
+
+The benefit of the client nodes should not be overstated though.
+Client nodes benefit other nodes by offloading the federated
+aspect of the request (e.g., the memory burden associated with
+federating a large aggregation). The routing aspect is a
+convenience, but the focus on any node should be on the burden
+that they offload from other nodes.
+
+To create a standalone data node:
+
+```yaml
+node.client: true
+```
+
+NOTE: It _is_ enough to just set `node.client: true`. This will
+overrule `node.master` and `node.data` settings.
+
+[float]
+[[node-configuration]]
+===== Node Configuration
+
+The above descriptions noted how to setup nodes to be master,
+data, or client nodes, but this section discusses some
+universal node settings that should be considered with any
+deployment.
+
+[float]
+[[node-configuration-disk]]
+====== Disk Configuration
+
+[cols="<,<,<",options="header",]
+|=======================================================================
+|Setting |Default |Description
+|`path.data` |`${ES_HOME}/data` |The place to store data (and metadata) for the
+node. This should be set uniquely per node.
+|`node.max_local_storage_nodes` |`50` |The number of nodes that can
+share the same `path.data` directory.
+|=======================================================================
+
+Every node should set its `path.data`, which, in the case of
+non-data nodes, allows the storage of metadata associated with
+the cluster. For data nodes, this is naturally where the data
+is stored.
+
+Whether you set the `path.data` or not, nodes can share
```

Comment:
I added a `WARNING:` to never run different node types from the same data path:

> WARNING: Never run different node types from the same data directory. This can lead to unexpected data loss.